### PR TITLE
KIM Integration - shoot name and networking for basic plans

### PR DESCRIPTION
**Pull request by jaroslaw-pieszka**
**Description**

Changes proposed in this pull request:

- Shoot name set for all plans
- Networking set for basic plans (Preview, AWS, GCP, Azure)
- Test assertions for networking added
- Control Plane Failure Tolerance 

**Related issue(s)**

#905

### CODE REVIEW

File: internal/process/provisioning/create_runtime_resource_step.go [hunk 1]
```diff
@@ -248,9 +244,55 @@ func (s *CreateRuntimeResourceStep) providerValues(operation *internal.Operation
 	return p.Provide(), nil
 }
 
+func (s *CreateRuntimeResourceStep) createHighAvailabilityConfiguration() *gardener.HighAvailability {
+
+	var failureTolerance *gardener.HighAvailability
+	if s.config.ControlPlaneFailureTolerance == string(gardener.FailureToleranceTypeZone) {
+		failureTolerance = &gardener.HighAvailability{
```
Comment (ralikio) [inline internal/process/provisioning/create_runtime_resource_step.go:251]: Can we shorten this method and only set "Type" field in the if block?

File: internal/process/provisioning/create_runtime_resource_step.go [hunk 2]
```diff
@@ -248,9 +244,55 @@ func (s *CreateRuntimeResourceStep) providerValues(operation *internal.Operation
...
+		networkingParams = &internal.NetworkingDTO{}
+	}
+
+	nodes := networkingParams.NodesCidr
+	if nodes == "" {
+		nodes = networking.DefaultNodesCIDR
+	}
+
+	return imv1.Networking{
+		Pods:     DefaultIfParamNotSet(networking.DefaultPodsCIDR, networkingParams.PodsCidr),
```
Comment (ralikio) [inline internal/process/provisioning/create_runtime_resource_step.go:279]: Is there any case in which DefaultPodsCIDR is not going to be set?

Comment (jaroslaw-pieszka) [reply internal/process/provisioning/create_runtime_resource_step.go:279]: I don't get you question.

File: internal/process/provisioning/create_runtime_resource_step.go (hunk 1 above)
Comment (jaroslaw-pieszka) [reply internal/process/provisioning/create_runtime_resource_step.go:251]: It will make it longer.
And which form is easier to read is highly subjective.

Comment (jaroslaw-pieszka) [reply internal/process/provisioning/create_runtime_resource_step.go:251]: In the next commit

File: internal/process/provisioning/create_runtime_resource_step.go (hunk 2 above)
Comment (ralikio) [reply internal/process/provisioning/create_runtime_resource_step.go:279]: Question removed.

File: internal/process/provisioning/create_runtime_resource_step.go [hunk 3]
```diff
@@ -248,9 +244,50 @@ func (s *CreateRuntimeResourceStep) providerValues(operation *internal.Operation
 	return p.Provide(), nil
 }
 
+func (s *CreateRuntimeResourceStep) createHighAvailabilityConfiguration() *gardener.HighAvailability {
+
+	failureType := gardener.FailureToleranceTypeZone
+	if s.config.ControlPlaneFailureTolerance != string(gardener.FailureToleranceTypeZone) {
+		failureType = gardener.FailureToleranceTypeNode
```
Comment (ralikio) [inline internal/process/provisioning/create_runtime_resource_step.go:251]: Can we change name to toleranceType?

**Review by ralikio — APPROVED** [review]

**Pull request merged**